impeller/entity/entity_pass_clip_stack.cc

```diff
@@ -145,6 +146,35 @@ EntityPassClipStack::ClipStateResult EntityPassClipStack::RecordClip(
     return result;
   }
 
+  // If the clip is an axis aligned rect and either is_aa is false or
+  // the clip is very nearly integral, then the depth write can be
+  // skipped for intersect clips. Since we use 4x MSAA, anything within
+  // about 0.2 of an integral value in either axis can be treated as
+  // approximately the same as an integral value.
+  if (!clip_coverage.is_difference_or_non_square &&
+      clip_coverage.coverage.has_value()) {
+    const Rect& coverage = clip_coverage.coverage.value();
+    constexpr Scalar threshold = 0.2;
```

Comment:
🤷‍♂️

Reply:
The [default 4x MSAA sample positions](https://developer.apple.com/documentation/metal/mtldevice/2866120-getdefaultsamplepositions#discussion) are 2/16 (0.125) pixels away from the edges. So maybe go with something a bit smaller, like 0.124?

![image](https://github.com/user-attachments/assets/12238701-9fe9-4621-8ca3-dd0aac624e32)